### PR TITLE
OCM-6753 | fix: hcp private cluster subnet validation

### PR DESCRIPTION
**Pull request by mikeshng**
In the PR https://github.com/openshift/rosa/pull/1854 

The `ValidateHostedClusterSubnets` publicSubnetsCount calculation were changed.
This cause rosa unable to create private hcp clusters in some cases due to validation failure:

```
E: The number of public subnets for a private hosted cluster should be zero
```

This change is to revert back to the previous calculation method. 

Reference: https://issues.redhat.com/browse/OCM-6753

### CODE REVIEW

**Comment by mikeshng** [comment]
CC @lucasponce @gbranco as discussed I am simply reverting based on my local testing. The original PR that made this change is a giant rebase PR that's hard to know the full context. Please if you have more knowledge around this area could you double check to make sure by reverting this change won't break other things?

/hold

Maintainers, please unhold when comfortable.

File: pkg/ocm/helpers.go
Comment (robpblake) [inline pkg/ocm/helpers.go:737]: @mikeshng Can we back this up with a unit test in `helpers_test.go`? 

I agree with your changes. I would like us to spend time now to write a test that ensures that we don't inadvertently break this again in the future as our lack of coverage is one of the major reasons why we have this type of regression.

If we add coverage as we go, it makes larger refactors easier in the future as we have a contract that tells us if we're breaking things.

Let me know if you want any help with the test. `Client` interface for AWS is mocked via `gomock` and available in `pkg/aws/mock_client.go`

cheers,

Rob

Comment (mikeshng) [reply pkg/ocm/helpers.go:737]: Hi @robpblake thank you for the review and feedback. I've done as you suggested. Could you PTAL again? Thanks.

File: pkg/ocm/helpers_test.go
Comment (den-rgb) [inline pkg/ocm/helpers_test.go:425]: I think this should return a `0` rather then a empty subnet array

**Review by robpblake — APPROVED** [review]
/lgtm

**Pull request merged**
